examples/quantization_2of4_sparse_w4a16/llama7b_sparse_w4a16.py

```diff
@@ -33,31 +33,61 @@
 bf16 = False  # using full precision for training
 lr_scheduler_type = "cosine"
 warmup_ratio = 0.1
-preprocessing_num_workers = 8
+preprocessing_num_workers = 64
 
-# this will run the recipe stage by stage:
-# oneshot sparsification -> finetuning -> oneshot quantization
-apply(
-    model=model,
+
+oneshot_kwargs = dict(
     dataset=dataset,
     recipe=recipe,
-    bf16=bf16,
-    output_dir=output_dir,
+    num_calibration_samples=num_calibration_samples,
+    preprocessing_num_workers=preprocessing_num_workers,
     splits=splits,
+    output_dir=output_dir,
+)
+
+training_kwargs = dict(
+    bf16=bf16,
```

Comment:
Why do we need this argument?

Reply:
This is a HF training argument
https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L722